test/Microsoft.IdentityModel.Tests/IdentityComparer.cs

```diff
@@ -37,108 +37,83 @@
 using Microsoft.IdentityModel.Tokens;
 using Microsoft.IdentityModel.Tokens.Saml;
 using Microsoft.IdentityModel.Xml;
+using System.Collections.ObjectModel;
+using System.Linq;
 
 namespace Microsoft.IdentityModel.Tests
 {
-    public class CompareContext
-    {       
-        List<string> _diffs = new List<string>();
+    public class IdentityComparer
+    {
+        private static readonly Dictionary<string, Func<Object, object, CompareContext, bool>> _equalityDict =
+            new Dictionary<string, Func<Object, object, CompareContext, bool>>()
+        {
+            {typeof(IEnumerable<Claim>).ToString(), AreClaimsEnumsEqual },
+            {typeof(IEnumerable<ClaimsIdentity>).ToString(), AreClaimsIdentitiesEnumsEqual },
+            {typeof(IEnumerable<string>).ToString(), AreStringEnumsEqual },
+            {typeof(IEnumerable<SecurityKey>).ToString(), AreSecurityKeyEnumsEqual },
+            {typeof(Newtonsoft.Json.Linq.JArray).ToString(), AreJArraysEqual },
+            {typeof(string).ToString(), AreStringsEqual },
+            {typeof(IDictionary<string, string>).ToString(), AreStringDictionariesEqual},
+            {typeof(Dictionary<string, object>).ToString(), AreStringObjectDictionariesEqual },
+            {typeof(Dictionary<string, object>.ValueCollection).ToString(), AreValueCollectionsEqual },
+            {typeof(IEnumerable<object>).ToString(), AreObjectEnumsEqual },
+            {typeof(Collection<SecurityKey>).ToString(), ContinueCheckingEquality },
+        };
+
+        // Checks if either object is null or if both are references of each other to see if we can terminate 
+        // comparison early. Returns null if neither case applies.
+        private static bool? AreObjectsNullOrReferences(Object o1, Object o2, CompareContext context)
```

Comment:
please change parameter names to object1, object2.
using English type names helps in readability.

(please change the other instances)